test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestResponse.java

```diff
@@ -80,7 +83,16 @@ public String getReasonPhrase() {
         List<String> warningHeaders = new ArrayList<>();
         for (Header header : response.getHeaders()) {
             if (header.getName().equals("Warning")) {
-                warningHeaders.add(header.getValue());
+                if (nodeVersion.after(Version.V_5_3_0_UNRELEASED) && response.getRequestLine().getMethod().equals("GET")
+                    && response.getRequestLine().getUri().contains("source")
+                    && response.getRequestLine().getUri().contains("source_content_type") == false && header.getValue().startsWith(
+                        "Deprecated use of the [source] parameter without the [source_content_type] parameter.")) {
```

Comment:
oh I see where the confusion is. Can you make this explanation a comment in there please?